airstack-modules/modules/airstack/nft-marketplace/nft.ts

```diff
@@ -1,184 +1,264 @@
+        createAirNftSaleTransaction(
+            chainId,
+            airBlock,
+            transactionHash,
+            logOrCallIndex,
+            sale.seller.toHexString(),
+            sale.buyer.toHexString(),
+            protocolType,
+            protocolActionType,
+            nftIds,
+            saleType,
+            sale.paymentToken.toHexString(),
+            sale.paymentAmount,
+            sale.protocolFees,
+            sale.protocolFeesBeneficiary.toHexString()
+        )
     }
 
+    /**
+     * @param buyer buyer address
+     * @param seller seller address
+     * @param nfts nfts class
+     *
+     */
     export class Sale {
         constructor(
             public buyer: Address,
             public seller: Address,
-            public nft: NFT,
+            public nfts: NFT[],
             public paymentAmount: BigInt,
             public paymentToken: Address,
             public protocolFees: BigInt,
             public protocolFeesBeneficiary: Address,
             public royalties: CreatorRoyalty[]
-        ) { }
+        ) {}
     }
 
+    /**
+     * @param fee royalty fee
+     * @param beneficiary royalty beneficiary
+     */
     export class CreatorRoyalty {
-        constructor(
-            public fee: BigInt,
-            public beneficiary: Address
-        ) { }
+        constructor(public fee: BigInt, public beneficiary: Address) {}
     }
 
+    /**
+     * @param collection nft collection address
+     * @param tokenId nft token id
+     * @param amount nft amount
+     */
     export class NFT {
         constructor(
             public readonly collection: Address,
-            public readonly standard: string, //ERC1155 or ERC721
             public readonly tokenId: BigInt,
             public readonly amount: BigInt
-        ) { }
+        ) {}
+    }
+}
+
+/**
+ *
+ * @param chainId chaind id
+ * @param block air block entity
+ * @param transactionHash transaction hash
+ * @param logOrCallIndex transaction log or call index
+ * @param from nft sender address
+ * @param to nft receiver address
+ * @param protocolType protocol type (eg: NFT_MARKET_PLACE)
+ * @param protocolActionType protocol action type (eg: BUY/SELL)
+ * @param nftIds AirNFT entity ids
+ * @param saleType sale type (eg: SINGLE_ITEM_SALE/BUNDLE_SALE)
+ * @param paymentToken payment token address used for sale
+ * @param paymentAmount payment amount for sale
+ * @param feeAmount protocol fee amount
+ * @param feeBeneficiary protocol fee beneficiary address
+ */
+function createAirNftSaleTransaction(
+    chainId: string,
+    block: AirBlock,
+    transactionHash: string,
+    logOrCallIndex: BigInt,
+    from: string,
+    to: string,
+    protocolType: string,
+    protocolActionType: string,
+    nftIds: string[],
+    saleType: string,
+    paymentToken: string,
+    paymentAmount: BigInt,
+    feeAmount: BigInt,
+    feeBeneficiary: string
+): void {
+    const id = chainId
```

Comment:
Id logic is duplicate, can we extract this into a function.

Reply:
done